collect/collect.go

```diff
@@ -289,6 +289,21 @@ func Sample(metric string, ts opentsdb.TagSet, v float64) error {
 	return nil
 }
 
+// StartTimer records the current time, and returns a function you can call to
+// record the end of your action.
+//
+// Typical usage would be:
+//    done := collect.StartTimer("myMetric", opentsdb.TagSet{})
+//    doMyThing()
+//    done()
+func StartTimer(metric string, ts opentsdb.TagSet) func() {
```

Comment:
Consider testing.Benchmark (http://stackoverflow.com/questions/8350609/how-do-you-time-a-function-in-go-and-return-its-runtime-in-milliseconds). Haven't looked into this in depth, maybe using testing here is just wrong. Up to you